### ISSUE TITLE
[Error] Flatpak Applications Fail to Start in Gnome Shell (Inside Container)

### ISSUE THREAD

**Issue by MR-TLL** [89luca89/distrobox#535]
**Describe the bug**
Hi all, i am using distrobox by running a full fledged Gnome Shell installed in a fedora toolbox 36 image, and using it as my main desktop with linux mint as host.

 Although Gnome does finds flatpak applications, clicking on the icons does nothing ( it loads for some time and then nothing happens) , interestingly it used to launch the first time i logged in, but doesn't work since then.

 Also running the flatpak app through terminal on host using flatpak run commands works flawlessly. (inside gnome shell on container)

But when i run the flatpak through terminal on distrobox container the following error comes up :- 
flatpak run com.usebottles.bottles
bwrap: Can't find source path /run/user/1000/doc/by-app/com.usebottles.bottles: No such file or directory

It would be really awesome to run flatpak apps through gnome shell, any ideas?

**To Reproduce**
Install Gnome shell using instructions on doc.
Run any flatpak apps installed on host using the desktop entry in gnome shell.

**Expected behavior**
The flatpak app doesn't starts.

**Logs**
NA

**Desktop (please complete the following information):**
 - Are you using podman or docker?  :- Podman
 - Which version or podman or docker? :- 3.4.4
 - Which version of distrobox? 1.4.1

**Additional context**
Containerized gnome shell also doesn't finds normal desktop entries, though i think this can be fixed by creating those entries manually, will create separate issue for that if needed.

**Comment by thjderjktyrjkt** [comment]
tbh running a full desktop environment inside distrobox is still largely untested,the existing documentations are based on others' blog posts. At a glance this seems related to https://github.com/89luca89/distrobox/issues/427 as xdg-document-portal mounts to `xdg-run/doc` as fuse

**Comment by 89luca89** [comment]
Hi @MR-TLL 

As @thjderjktyrjkt said, running a full fledged DE inside distrobox is experimental and food for thoughts :smile: 

And on top of that running stuff like Flatpak (or Snap) inside distrobox is not supported (yet?)

Try using `distrobox-host-exec` for flatpaks, remove flatpak package from inside the distrobox and do a 
`sudo ln -s /usr/bin/distrobox-host-exec /usr/bin/flatpak`, so that `flatpak` commands are issued on the host instead of the guest

**Comment by MR-TLL** [comment]
Hi @89luca89 sorry for the late reply...
Any plans to integrate running DE more seamless? TBH this will be a killer feature of distrobox, and something i am very excited to use as daily driver if possible :smile: 
Actually i don't want to run flatpak inside the container, i just want the containerized DE to be able to detect and start host flatpak apps on the host itself... which it can't seems to do a the moment.
TLDR :- The flatpak apps i want to run are inside the host and i want them to be run on the host itself, i just want the ability for the containerized DE to start them ( like clicking on the flatpak icon on gnome app list) (instead of starting from terminal :( )

**Comment by 89luca89** [comment]
> Any plans to integrate running DE more seamless? TBH this will be a killer feature of distrobox, and something i am very excited to use as daily driver if possible 

To be honest, no, I don't have this much free time :sweat_smile:  and there is many other stuff to do that I'd put in a higher priority than this

> TLDR :- The flatpak apps i want to run are inside the host and i want them to be run on the host itself, i just want the ability for the containerized DE to start them ( like clicking on the flatpak icon on gnome app list) (instead of starting from terminal :( )

Inside the container, create a link with `distrobox-host-exec` 

`sudo ln -s /usr/bin/distrobox-host-exec /usr/bin/flatpak` or `/usr/local/bin/flatpak` whichever works for you

This will make the `flatpak` binary be executed on the host, and you'll be able to use `flatpak run` to start host's stuff

**Comment by profetik-777** [comment]
> Hi @89luca89 sorry for the late reply...
> 
> Any plans to integrate running DE more seamless? TBH this will be a killer feature of distrobox, and something i am very excited to use as daily driver if possible smile
> 
> Actually i don't want to run flatpak inside the container, i just want the containerized DE to be able to detect and start host flatpak apps on the host itself... which it can't seems to do at the moment.
> 
> TLDR :- The flatpak apps i want to run are inside the host and i want them to be run on the host itself, i just want the ability for the containerized DE to start them ( like clicking on the flatpak icon on gnome app list) (instead of starting from terminal :( )

@89luca89 should we close this one since we haven't heard anything back?

**Comment by 89luca89** [comment]
Yea closing as not planned
